### ISSUE TITLE
Receiving ConcurrentModificationException from ServerRequest.getPostWithInstrumentationValues()

### ISSUE THREAD

**Issue by petrusgomes** [BranchMetrics/Android-Deferred-Deep-Linking-SDK#198]
We're receiving exceptions on Crashlytics with the following stacktrace:

```java
Fatal Exception: java.lang.RuntimeException: An error occured while executing doInBackground()
       at android.os.AsyncTask$3.done(AsyncTask.java:300)
       at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:355)
       at java.util.concurrent.FutureTask.setException(FutureTask.java:222)
       at java.util.concurrent.FutureTask.run(FutureTask.java:242)
       at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
       at java.lang.Thread.run(Thread.java:841)
Caused by java.util.ConcurrentModificationException
       at java.util.HashMap$HashIterator.nextEntry(HashMap.java:806)
       at java.util.HashMap$KeyIterator.next(HashMap.java:833)
       at io.branch.referral.ServerRequest.getPostWithInstrumentationValues(ServerRequest.java:166)
       at io.branch.referral.Branch$BranchPostTask.doInBackground(Branch.java:3337)
       at io.branch.referral.Branch$BranchPostTask.doInBackground(Branch.java:3313)
       at android.os.AsyncTask$2.call(AsyncTask.java:288)
       at java.util.concurrent.FutureTask.run(FutureTask.java:237)
       at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
       at java.lang.Thread.run(Thread.java:841)
```

Are we doing something wrong? It seems some bug on the SDK. Our app is using basic Branch deeplink implementation, with branchUniversalObject.generateShortUrl()

Thank you

**Comment by sojanpr** [comment]
@petrusgomes Thank for the detailed  reported. This issue was present in SDK v1.10.2 and is already fixed with v 1.10.3 . 
Could you please try with latest version (1.10.3)

**Comment by petrusgomes** [comment]
Great! I'm using

```compile 'io.branch.sdk.android:library:1.+'```` so my apk should be using the newest release... I will double check.
Can you detail how many users may be affected by this? I could not reproduce the error, but it is occurring in production env...

**Comment by sojanpr** [comment]
This was corner case  present only in release 1.10.2 which is released on Dec -15 - 2015. So only releases that made with `compile 'io.branch.sdk.android:library:1.+'` before Dec-15 will not have this issue

**Comment by petrusgomes** [comment]
I see. But I think this issue should be revisited, because my build is from today and I checked that is using version 1.10.3.

Thanks for your support and fast response

**Comment by sojanpr** [comment]
@petrusgomes we are bale to reproduce the issue. Will soon open a PR for this

**Comment by sojanpr** [comment]
@petrusgomes Opened a new PR for this issue
https://github.com/BranchMetrics/Android-Deferred-Deep-Linking-SDK/pull/199
DO you think this is will solve the issue. It is working fine with our tests. Will push this to maven soon

**Comment by petrusgomes** [comment]
Thank you. The PR seems to solve it.

**Comment by sojanpr** [comment]
@petrusgomes We have released 1.10.4 with the fix. Thank you very much.
